platform/core/PolyPodCoreSwift/Sources/PolyPodCoreSwift/Core.swift

```diff
@@ -0,0 +1,85 @@
+import PolyPodCore
+import Foundation
+import FlatBuffers
+
+/// Possible errors that can be thrown by PolyPodCoreSwift
+public enum PolyPodCoreError: Error {
+    /// Internal Rust Core failure with error code and message
+    case internalCoreFailure(Failure)
+    /// Rust Core returned an invalid result type for a given operation
+    case invalidResult(String)
+    /// Rust Core returned an invalid failure content
+    case invalidFailure(String)
+    
+    var localizedDescription: String {
+        switch self {
+        case let .internalCoreFailure(content):
+            return "Internal Core Failure: \(content.code) \(String(describing: content.message))"
+        case let .invalidResult(content):
+            return "Core Failure: \(content)"
+        case let .invalidFailure(content):
+            return "Core Failure: \(content)"
+        }
+    }
+}
+
+/// Swift wrapper around the Rust Core. Encapsulates specific Pointer and Flabuffer operations.
+public final class Core {
+    public static let instance = Core()
+    
+    // MARK: - Private config
+    private var languageCode: UnsafePointer<CChar>!
+    
+    private init() {}
+    
+    // MARK: - Public API
+    
+    /// Prepares the core to be used
+    /// Should be called before invoking any other API
+    public func bootstrap(languageCode: String) -> Result<Void, Error> {
+        // Force unwrap should be safe
+        self.languageCode = NSString(string: languageCode).utf8String!
+        
+        return processCoreResponse(core_bootstrap(languageCode)) { byteBuffer in
+            let response = CoreBootstrapResponse.getRootAsCoreBootstrapResponse(bb: byteBuffer)
+            if let failure = response.failure {
+                throw PolyPodCoreError.internalCoreFailure(failure)
+            }
+        }
+    }
+    
+    /// Parse the FeatureManifest from the given json
+    /// - Parameter json: Raw JSON to parse the FeatureManifest from
+    /// - Returns: A FeatureManifest if parsing succeded, nil otherwise
+    public func parseFeatureManifest(json: String) -> Result<FeatureManifest, Error> {
+        processCoreResponse(parse_feature_manifest_from_json(json)) { byteBuffer in
+            let response = FeatureManifestParsingResponse.getRootAsFeatureManifestParsingResponse(bb: byteBuffer)
+            switch response.resultType {
+            case .featuremanifest:
+                return response.result(type: FeatureManifest.self)
+            case .failure:
+                if let failure = response.result(type: Failure.self) {
+                    throw PolyPodCoreError.internalCoreFailure(failure)
+                } else {
```

Comment:
Nit: Since the previous `throw` would exit program flow, no `else` statement would be needed (assuming I understand Swift exceptions sufficiently well)

Reply:
This branch will be executed if `response.result(type: Failure.self)` returns `nil`.

Reply:
I know, but my point is: If the `throw PolyPodCoreError.internalCoreFailure(failure)` happens, this _cannot_ happen. If it doesn't happen, this will _definitely_ happen. So the code above is equivalent to (the shorter):

```
if let failure = response.result(type: Failure.self) {
    throw PolyPodCoreError.internalCoreFailure(failure)
}
throw PolyPodCoreError.invalidFailure(...)
```

But it's just a nit, I'm a fan of "no `else` after `return` or `throw`" code style rules, since I believe it makes the code cleaner and avoids the arrow anti pattern, but we don't have such a rule in place right now, so it doesn't matter.

Reply:
Oh, ok, miss-understood your initial Nit.
I am more on a side `if-else`, to avoid the case when the last `return` or `throw` is used as generic fallback(when new conditions are added, like another `if`), while it was needed originally intended.